src/Payment.ts

```diff
@@ -59,12 +59,14 @@ export class Payment {
    * @param options.gasPrice Custom gas price.
    * @param options.gasLimit Custom gas limit.
    * @param options.feePayer Either `sender` or `receiver`. Specifies who pays network fees.
+   * @param extraData Extra data that will appear in the Transfer event when successful
    */
   public async prepare(
     networkAddress: string,
     receiverAddress: string,
     value: number | string,
-    options: PaymentOptions = {}
+    options: PaymentOptions = {},
+    extraData: string = '0x'
```

Comment:
This is better part of `PaymentOptions` then you dont have to do the `undefined` hack you do in the e2e tests. 
There are no named parameters in typescript, so we use the object approach. This should be the last argument.
See https://medium.com/@afontcu/cool-javascript-9-named-arguments-functions-that-get-and-return-objects-337b6f8cfa07 for mor info.

Reply:
Fixed in other PR